flint/flint-spark-integration/src/main/scala/org/apache/spark/sql/flint/FlintTable.scala

```diff
@@ -27,14 +30,20 @@ case class FlintTable(conf: util.Map[String, String], userSpecifiedSchema: Optio
     with SupportsRead
     with SupportsWrite {
 
-  val name = FlintSparkConf(conf).tableName()
+  lazy val flintSparkConf: FlintSparkConf = FlintSparkConf(conf)
+
+  lazy val name: String = flintSparkConf.tableName()
 
   var schema: StructType = {
     if (schema == null) {
       schema = if (userSpecifiedSchema.isDefined) {
         userSpecifiedSchema.get
       } else {
-        throw new UnsupportedOperationException("infer schema not supported yet")
+        FlintDataType.deserialize(
+          FlintClientBuilder
+            .build(flintSparkConf.flintOptions())
+            .getIndexMetadata(name)
+            .getContent)
```

Comment:
just wonder is the schema infer logic here also needed by DataFrame write?

Reply:
for example, when write df to json. DateTimeType field should be formatted. The format should align with the format in the OpenSearch index mapping.
more reading https://github.com/opensearch-project/sql/blob/feature/flint/flint/flint-spark-integration/src/main/scala/org/apache/spark/sql/flint/json/FlintJacksonGenerator.scala#L119